addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/server/BinRpcResponseHandler.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homematic.internal.communicator.server;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.net.Socket;
+
+import org.openhab.binding.homematic.internal.common.HomematicConfig;
+import org.openhab.binding.homematic.internal.communicator.message.BinRpcMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads a BIN-RPC message from the socket and handles the method call.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class BinRpcResponseHandler implements Runnable {
+    private final Logger logger = LoggerFactory.getLogger(BinRpcResponseHandler.class);
+
+    private Socket socket;
+    private RpcResponseHandler<byte[]> rpcResponseHandler;
+    private HomematicConfig config;
+
+    public BinRpcResponseHandler(Socket socket, RpcResponseHandler<byte[]> rpcResponseHandler, HomematicConfig config) {
+        this.socket = socket;
+        this.rpcResponseHandler = rpcResponseHandler;
+        this.config = config;
+    }
+
+    /**
+     * Reads the event from the Homematic gateway and handles the method call.
+     */
+    @Override
+    public void run() {
+        try {
+            BinRpcMessage message = new BinRpcMessage(socket.getInputStream(), true, config.getEncoding());
+            if (logger.isTraceEnabled()) {
+                logger.trace("Event BinRpcMessage: {}", message.toString());
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies. 
In that case the isTraceEnabled is not needed anymore.